config/ompi_setup_mpi_fortran.m4

```diff
@@ -413,8 +413,8 @@ end program]])],
            AS_IF([test $OMPI_FORTRAN_HAVE_IGNORE_TKR -eq 1],
                  [OMPI_FORTRAN_USEMPI_DIR=mpi/fortran/use-mpi-ignore-tkr
                   OMPI_FORTRAN_USEMPI_LIB=-l${with_libmpi_name}_usempi_ignore_tkr],
-                 [OMPI_FORTRAN_USEMPI_DIR=mpi/fortran/use-mpi-tkr
-                  OMPI_FORTRAN_USEMPI_LIB=-l${with_libmpi_name}_usempi])
+                 [AC_MSG_WARN([** Fortran compiler does not support ignoring tkr.  Please use a newer fortran compiler])
+                  AC_MSG_ERROR([*** Cannot continue])])
```

Comment:
A few thoughts here...

1. Can we update the comment that comes before this, too?  It talks about deciding whether to use the TKR or ignore-TKR version.

2. Extra bonus points for totally ditching `OMPI_FORTRAN_USEMPI_DIR` and `OMPI_FORTRAN_USEMPI_LIB` throughout the rest of the code base.  It isn't strictly necessary to remove it, but it will now only ever be a single value.  While technically not actually needed for this PR, it might be nice to clean that up.

3. Do all modern Fortran compilers support ignore TKR?  E.g., are there any purist Fortran compilers that support what we need for `mpi_f08`, but do *not* support ignore TKR?  Specifically: is there ever a case where we could build the `mpif.h` stuff and the `mpi_f08` module, but *not* the ignore-TKR `mpi` module?

Reply:
for 1 yes didn't notice that.  sure 2 sounds good as well.
For 3 only possible one that I know if may be NAG.  But I don't have access to that compiler.

Reply:
@jsquyres check now.

Reply:
@jsquyres ping